pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -835,6 +836,72 @@ func TestUpdateTaskRunsState(t *testing.T) {
 
 }
 
+func TestUpdateTaskRunsWithWhenExpressions(t *testing.T) {
+	// TestUpdateTaskRunsWithWhenExpressions runs "getTaskRunsStatus" and verifies how it updates a PipelineRun status
+	// from a TaskRun with When Expressionsassociated to the PipelineRun
+	pr := tb.PipelineRun("test-pipeline-run", tb.PipelineRunNamespace("foo"), tb.PipelineRunSpec("test-pipeline"))
+	pipelineTask := v1beta1.PipelineTask{
+		Name:    "unit-test-1",
+		TaskRef: &v1beta1.TaskRef{Name: "unit-test-task"},
+		WhenExpressions: []v1beta1.WhenExpression{{
+			Input:    "foo",
+			Operator: selection.NotIn,
+			Values:   []string{"foo", "bar"},
```

Comment:
what do you think about including some variable replacement here as well to get the full picture?

Reply:
added them below, with parameter and results replacements, with both evaluating to true and evaluating to false